### PR TITLE
Circle detection testing branch

### PR DESCRIPTION
**Pull request by pauliuskalv**
Hello, I added Object tracking by using the EmguCV Tracker class. Can you review my code?

### CODE REVIEW

File: TOPBanga/Detection/DetectionMain.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        public static Mat frame;
+        private System.Threading.Timer timer;
+        public DetectionMain( String videoFileName )
+        {
+            tracker = new Tracker("KCF");
+            video = new VideoCapture(videoFileName);
+            video.Grab();
+            frame = video.QueryFrame();
+            Init(new Point(580,520));
+            timer = new System.Threading.Timer(this.CalcImage, null, 0, 60);
```
Comment (solarbeam) [inline TOPBanga/Detection/DetectionMain.cs]: These lines have hard coded values: either put them as constants or put a comment explaining them.

File: TOPBanga/MainForm.cs [hunk 2]
```diff
@@ -5,144 +5,75 @@
...
-            System.GC.Collect();
-        } 
         private void BrowseButton_Click(object sender, EventArgs e)
         {
             OpenFileDialog Openfile = new OpenFileDialog() {
-                /*
-                 * TODO
-                 *  add more extensions
-                 */
                 Filter = "PNG file|*.png|JPG file|*.jpg|MP4 file|*.mp4|File|*.*",
```
Comment (solarbeam) [inline TOPBanga/MainForm.cs]: If we are tracking only videos now, there is no need for image file extensions in this line.

File: TOPBanga/MainForm.cs [hunk 3]
```diff
@@ -5,144 +5,75 @@
...
                 InitialDirectory = "C:\\"
```
Comment (solarbeam) [inline TOPBanga/MainForm.cs]: If you leave this string empty the browser will open in the working directory; right where the video is.

File: TOPBanga/MainForm.cs [hunk 4]
```diff
@@ -5,144 +5,75 @@
...
-                if ( extension.Contains("png") || extension.Contains("jpg") )
-                {
-                    image = new Image<Bgr, Byte>(Openfile.FileName);
-                    calcImage();
-                    return;
-                }
-
-                video = new VideoCapture(Openfile.FileName);
-                timer = new System.Threading.Timer(this.calcFrame, null, 0, 100);
+                detect = new DetectionMain(Openfile.FileName);
```
Comment (solarbeam) [inline TOPBanga/MainForm.cs]: Use "this" if possible.

File: TOPBanga/MainForm.cs [hunk 5]
```diff
@@ -5,144 +5,75 @@
...
+                detect.FrameChange += new EventHandler(this.frameChange);
             }
         }
 
+        public void frameChange (object sender, EventArgs e)
+        {
+            Console.WriteLine("AAA");
```
Comment (solarbeam) [inline TOPBanga/MainForm.cs]: I suggest you either remove this line, or make it printout something more meaningful like "Frame" or somthing.

File: TOPBanga/MainForm.cs [hunk 6]
```diff
@@ -5,144 +5,75 @@
...
+            Picture.Image = DetectionMain.frame.Bitmap;
+        }
+
         private void Refresh_Button_Click(object sender, EventArgs e)
```
Comment (solarbeam) [inline TOPBanga/MainForm.cs]: If the following event functions are not used I suggest to remove them.

**Comment by pauliuskalv** [comment]
@mrzilvis , I see that the Tracker does not track as expected, so I chose to close this pull request and make the necessary optimizations later.

**Pull request closed without merging**